### PR TITLE
ENH: pcds-4.1.5

### LINKED ISSUE #190: torch for UED
A UED scientist said that they needed torch, gyptorch, and botorch. I dropped these into the ued-dev environment that we've been using for the GUI.

But these should go into the standard environment as well:
- `pytorch`
- `gpytorch`
- `botorch`

### PR DESCRIPTION
**Pull request by ZLLentz**
- Temporarily switch channel priority to flexible for active conda-forge CDN issues
- Add a check and a warning when conda-forge isn't ready for a package
- pin `pytest-qt` back to `3.3.0`. The `4.0.0` release has some API changes that break the `pydm` and `typhos` tests.
- closes #190 

PCDS package updates:
|  Package | Old | New | Release Notes |
| - | - | - | - |
| ads-async | 1.1.0 | 1.1.1 | https://github.com/pcdshub/ads-async/releases/tag/v0.1.1 |
| hutch-python | 1.10.0 | 1.10.1 | https://github.com/pcdshub/hutch-python/releases/tag/v1.10.1 |
| pcdscalc | 0.3.0 | 0.3.1 | https://github.com/pcdshub/pcdscalc/releases/tag/v0.3.1 |
| pcdsdevices | 4.4.0 | 4.5.0 | https://github.com/pcdshub/pcdsdevices/releases/tag/v4.5.0 |
| pmgr | 1.1.4 | 2.0.2 | https://github.com/pcdshub/pmgr/releases/tag/R2.0.2 https://github.com/pcdshub/pmgr/releases/tag/R2.0.1 |

Lab community package notable updates:
|  Package | Old | New |
| - | - | - |
| caproto | 0.8.0 | 0.8.1 |
| epicscorelibs | 7.0.3.99.4.0 | 7.0.4.99.1.2 |
| pyepics | 3.4.3 | 3.5.0 |

Python community package notable updates:
|  Package | Old | New |
| - | - | - |
| botorch | N/A | 0.4.0 |
| gpytoch | N/A | 1.4.2 |
| ipython | 7.23.0 | 7.24.1 |
| pytorch | N/A | 1.8.0 |

Original issue text from last night:

This is a bit broken because it seems to take a long long time for pcdsdevices to become available on conda-forge.
I am able to download the tarball from the anaconda website and create the environment from a local mirror but am unable to install from the cli directly from conda-forge.

I've added a warning message to the scripts and am considering a full stop of the build if this issue is detected, since it causes terrible issues with no reasonable error messages.

I will revisit tomorrow when things are more stable and retry doing #190, I removed the packages because I assumed they were the issue.

### CODE REVIEW

**Comment by ZLLentz** [comment]
Test failure and segfault also happens locally with this env- looking more closely

**Comment by klauer** [comment]
Comment from the sidelines: really, really nice PR description format!

**Comment by ZLLentz** [comment]
The parts of the travis required to pass have passed! :tada:

**Review by klauer — APPROVED** [review]
Remember how Jinja2 was failing our docs build with ads-deploy? Doing a manual check, it's good that jinja didn't get updated to the incompatible version: https://github.com/pcdshub/ads-deploy/blob/master/requirements.txt
We don't rely on pcds-envs for that, but even so - I'd imagine it might break something else with our heavy usage of jinja.

Everything else looks good, with no incompatibilities I'm aware of. 👍

File: envs/pcds/conda-packages.txt [hunk 1]
```diff
@@ -1,9 +1,10 @@
-ads-async>=0.1.0
+ads-async>=0.1.1
 anaconda-client
 bluesky>=1.6.7
 bluesky-live
 bokeh>=1.0.1
-caproto>=0.8.0
+botorch
```
[the remark this answers is not in the document]
Comment (ZLLentz) [reply envs/pcds/conda-packages.txt:6]: :shrug:

**Comment by ZLLentz** [comment]
I guess there must be at least one package that requests an older jinja version?
I'm happy to pin it back if needed.

**Comment by ZLLentz** [comment]
I think setting the channel priority to flexible has fixed the dev/master env build, it may be worth forcing it to flexible for that build particularly even if we set it back to strict for all other builds.

**Comment by klauer** [comment]
> I guess there must be at least one package that requests an older jinja version?  I'm happy to pin it back when it becomes needed.
Yeah, I think that's fine. It'd be good to keep in the back of our minds going forward, at least. I worry a bit about our test coverage in places (pytmc, looking at you).

> I think setting the channel priority to flexible has fixed the dev/master env build, it may be worth forcing it to flexible for that build particularly even if we set it back to strict for all other builds.
That's great. Seems very reasonable for dev, at the very least.

**Comment by ZLLentz** [comment]
All packages pass in dev except `transfocate`, which we know needs a minor testing update.

**Pull request merged**